File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <bits/stdc++.h>
using namespace std;

string S, T, ret;
stack<char> stk;

int main() {
  cin >> S >> T;

  for (char a : S) {
    stk.push(a);
    if (stk.size() >= T.size() && stk.top() == T[T.size() - 1]) {
      string s = "";
      for (int i = 0; i < T.size(); i++) {
        s += stk.top();
        stk.pop();
      }
      reverse(s.begin(), s.end());
      if (s != T) {
        for (char b : s) stk.push(b);
      }
    }
  }
  if (stk.empty())
    cout << "FRULA";
  else {
    while (!stk.empty()) {
      ret += stk.top();
      stk.pop();
    }
    reverse(ret.begin(), ret.end());
    cout << ret;
  }
  return 0;
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : './input.txt').toString().trim().split('\n');

const [s, target] = input;

function answer(){
    const stk = [];

    for (let i = 0; i < s.length; i++){
        stk.push(s[i]);
        
        if(stk.length >= target.length && stk[stk.length - 1] === target[target.length - 1]){
            let str = '';
            for (let j = 0; j < target.length; j++){
                str += stk[stk.length - 1];
                stk.pop();
            }

            str = str.split('').reverse().join('');

            if(target !== str){
                for (let k = 0; k < str.length; k++){
                    stk.push(str[k]);
                }
            }
        }
    }

    console.log(stk.length ? stk.join('') : 'FRULA');
}

answer();